### PR TITLE
Manual updates 20231028

### PR DESCRIPTION
**Pull request by moljac**
### Does this change any of the generated binding API's?

Yes.

Mostly nuget strucutre.

- `Xamarin.Chromium.CroNet`

   removed properties

### Describe your contribution

- `com.google.android.gms:playservices-cloudmessaging` - 17.0.2 -> 17.1.0
- `com.google.android.gms:playservices-pal` - 20.1.1 -> 20.2.0
- `com.google.android.gms:playservices-pay` - 16.3.0 -> 16.4.0
- `com.google.android.play:integrity` - 1.2.0 -> 1.3.0
- `com.google.firebase:firebase-auth` - 22.2.0 -> 22.3.0
- `com.google.firebase:firebase-auth-ktx` - 22.2.0 -> 22.3.0
- `com.google.firebase:firebase-config` - 21.5.0 -> 21.6.0
- `com.google.firebase:firebase-config-ktx` - 21.5.0 -> 21.6.0
- `com.google.firebase:firebase-config-interop` -  -> 16.0.0
- `com.google.firebase:firebase-crashlytics` - 18.5.1 -> 18.6.0
- `com.google.firebase:firebase-crashlytics-ktx` - 18.5.1 -> 18.6.0
- `com.google.firebase:firebase-crashlytics-ndk` - 18.5.1 -> 18.6.0
- `com.google.firebase:firebase-perf` - 20.5.0 -> 20.5.1
- `com.google.firebase:firebase-perf-ktx` - 20.5.0 -> 20.5.1
- `com.google.firebase:firebase-sessions` - 1.1.0 -> 1.2.0
- `com.google.mlkit:common` - 18.9.0 -> 18.10.0
- `com.google.mlkit:imagelabeling-custom` - 17.0.1 -> 17.0.2
- `com.google.mlkit:objectdetection-custom` - 17.0.0 -> 17.0.1
- `com.google.mlkit:translate` - 17.0.1 -> 17.0.2
- `com.google.mlkit:vision-interfaces` - 16.2.0 -> 16.3.0
- `com.google.android.libraries.places:places` - 3.2.0 -> 3.3.0
- `com.android.billingclient:billing` - 6.0.1 -> 6.1.0
- `com.android.billingclient:billing-ktx` - 6.0.1 -> 6.1.0
- `com.google.android.datatransport:transport-api` - 3.0.0 -> 3.1.0
- `com.google.android.datatransport:transport-backend-cct` - 3.1.9 -> 3.2.0
- `com.google.android.datatransport:transport-runtime` - 3.1.9 -> 3.2.0
- `com.google.protobuf:protobuf-javalite` - 3.24.4 -> 3.25.1
- `org.chromium.net:cronet-api` - 108.5359.79 -> 113.5672.61
- `org.chromium.net:cronet-common` - 108.5359.79 -> 113.5672.61
- `org.chromium.net:cronet-embedded` - 108.5359.79 -> 113.5672.61
- `org.chromium.net:cronet-fallback` - 108.5359.79 -> 113.5672.61
- `com.google.android.gms:playservices-ads-lite` - 22.3.0 -> 20.0.0

### CODE REVIEW

**Comment by moljac** [comment]
Missing pacakges

```
Dependency errors : 5
1
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-ads:22.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


2
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices-java:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-ads:22.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices-java",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


3
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-measurement-impl:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


4
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices-java:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-measurement-impl:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices-java",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


5
	System.Exception: 
No matching artifact config found for: 
			androidx.credentials.credentials:1.0.0-alpha04
to satisfy dependency of: 
			com.google.android.libraries.identity.googleid.googleid:1.1.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.credentials",
        "artifactId": "credentials",
        "version": "1.0.0-alpha04",
        "nugetVersion": "CHECK PREFIX 1.0.0-alpha04",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
```

**Comment by moljac** [comment]
```
Dependency errors : 4
1
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-ads:22.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


2
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices-java:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-ads:22.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices-java",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


3
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-measurement-impl:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


4
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices-java:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-measurement-impl:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices-java",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
```

**Comment by moljac** [comment]
https://github.com/xamarin/AndroidX/pull/826

**Comment by moljac** [comment]
Update downgrade attempt:

```
Dependency errors : 7
1
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-measurement-impl:21.4.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


2
	System.Exception: 
No matching artifact config found for: 
			androidx.privacysandbox.ads.ads-adservices-java:1.0.0-beta05
to satisfy dependency of: 
			com.google.android.gms.play-services-measurement-impl:21.4.0

	Please add following json snippet to config.json:


      {
        "groupId": "androidx.privacysandbox.ads",
        "artifactId": "ads-adservices-java",
        "version": "1.0.0-beta05",
        "nugetVersion": "CHECK PREFIX 1.0.0-beta05",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


3
	System.Exception: 
No matching artifact config found for: 
			com.google.android.gms.play-services-ads:22.5.0
to satisfy dependency of: 
			com.google.firebase.firebase-ads:22.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-ads",
        "version": "22.5.0",
        "nugetVersion": "CHECK PREFIX 22.5.0",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


4
	System.Exception: 
No matching artifact config found for: 
			com.google.android.gms.play-services-ads-lite:22.5.0
to satisfy dependency of: 
			com.google.firebase.firebase-ads-lite:22.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-ads-lite",
        "version": "22.5.0",
        "nugetVersion": "CHECK PREFIX 22.5.0",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


5
	System.Exception: 
No matching artifact config found for: 
			com.google.android.gms.play-services-measurement:21.5.0
to satisfy dependency of: 
			com.google.firebase.firebase-analytics:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-measurement",
        "version": "21.5.0",
        "nugetVersion": "CHECK PREFIX 21.5.0",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


6
	System.Exception: 
No matching artifact config found for: 
			com.google.android.gms.play-services-measurement-api:21.5.0
to satisfy dependency of: 
			com.google.firebase.firebase-analytics:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-measurement-api",
        "version": "21.5.0",
        "nugetVersion": "CHECK PREFIX 21.5.0",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
						


7
	System.Exception: 
No matching artifact config found for: 
			com.google.android.gms.play-services-measurement-sdk:21.5.0
to satisfy dependency of: 
			com.google.firebase.firebase-analytics:21.5.0

	Please add following json snippet to config.json:


      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-measurement-sdk",
        "version": "21.5.0",
        "nugetVersion": "CHECK PREFIX 21.5.0",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }

```

File: source/com.google.android.datatransport/transport-runtime/Additions/Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.ProviderOfLazy.cs [hunk 1]
```diff
@@ -13,15 +13,15 @@ public sealed partial class ProviderOfLazy //: global::Java.Lang.Object, global:
...
-		public unsafe global::Java.Lang.Object Get ()
-		{
-			const string __id = "get.()Lcom/google/android/datatransport/runtime/dagger/Lazy;";
-			try {
-				var __rm = _members.InstanceMethods.InvokeAbstractObjectMethod (__id, this, null);
-				return (Java.Lang.Object) global::Java.Lang.Object.GetObject<global::Xamarin.Google.Android.DataTransport.Runtime.Dagger.ILazy> (__rm.Handle, JniHandleOwnership.TransferLocalRef);
-			} finally {
-			}
-		}
+		// [Register ("get", "()Lcom/google/android/datatransport/runtime/dagger/Lazy;", "")]
```
Comment (jpobst) [inline source/com.google.android.datatransport/transport-runtime/Additions/Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.ProviderOfLazy.cs:16]: Looking at the diff for this class:

    #### Type Changed: Xamarin.Google.Android.DataTransport.Runtime.Dagger.Internal.ProviderOfLazy

    Modified base type:

    ```diff
    -Java.Lang.Object
    +System.Object
    ```

    Added constructor:

    ```csharp
    public ProviderOfLazy ();
    ```

    Removed interfaces:

    ```csharp
    Android.Runtime.IJavaObject
    Java.Interop.IJavaPeerable
    JavaX.Inject.IProvider
    System.IDisposable
    ```

    Removed properties:

    ```csharp
    public override Java.Interop.JniPeerMembers JniPeerMembers { get; }
    protected override IntPtr ThresholdClass { get; }
    protected override System.Type ThresholdType { get; }
    ```

    Removed methods:

    ```csharp
    public static JavaX.Inject.IProvider Create (JavaX.Inject.IProvider provider);
    public virtual Java.Lang.Object Get ();
    ```

I suspect the entire class was removed, and we should remove this entire partial class.  Same with `SetFactory`.

File: config.json [hunk 2]
```diff
@@ -2563,19 +2570,11 @@
...
-        "nugetVersion": "122.3.0",
-        "nugetId": "Xamarin.GooglePlayServices.Ads.Base",
-        "dependencyOnly": true
-      },
       {
         "groupId": "com.google.android.gms",
         "artifactId": "play-services-ads-lite",
-        "version": "22.3.0",
-        "nugetVersion": "122.3.0",
+        "version": "20.0.0",
```
[the remark this answers is not in the document]
Comment (moljac) [reply config.json:2576]:
```json
      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-ads-lite",
        "version": "20.0.0",
        "nugetVersion": "120.0.0",
        "nugetId": "Xamarin.GooglePlayServices.Ads.Lite",
        "dependencyOnly": true
      }
```

Dependency. I cleaned up dependencies (extras, duplicates, stale versions etc). This dependency solves multiple relations.

[the remark this answers is not in the document]
Comment (moljac) [reply config.json:2580]: https://github.com/xamarin/GooglePlayServicesComponents/pull/821/files#diff-587cb980af76fdc7e52369fd0b9d926dff266976b6f8ac631e358fecc49ff8cfL2754-L2761

[the remark this answers is not in the document]
Comment (moljac) [reply config.json:2580]: with 22.3.0

```
Dependency errors : 1
1
	System.Exception: 
No matching artifact config found for: 
			com.google.android.gms.play-services-ads-lite:[20.0.0]
to satisfy dependency of: 
			com.google.android.gms.play-services-gass:20.0.0

	Please add following json snippet to config.json:

      {
        "groupId": "com.google.android.gms",
        "artifactId": "play-services-ads-lite",
        "version": "[20.0.0]",
        "nugetVersion": "CHECK PREFIX [20.0.0]",
        "nugetId": "CHECK NUGET ID",
        "dependencyOnly": true/false
      }
```

**Review by jpobst — APPROVED** [review]

**Pull request merged**